### PR TITLE
feat(openapi): support const (#521)

### LINKED ISSUE #521: using `const` keyword produce invalid swagger file
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.25.0

### Plugin version

4.13.0

### Node.js version

16.13.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

10.15.7 (19H2)

### Description

The swagger plugin produces an invalid swagger file when using the json-schema `const` keyword ([spec](http://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.6.1.3))

### Steps to Reproduce

```
const requestBodySchema = {
  type: 'object',
  required: ['prop'],
  properties: {
    prop: { const: 'my-const' },
  },
}

export default async (fastify) => {
  fastifty.route({
  method: 'POST',
  url: '/',
  schema: {
    body: requestBodySchema,
  },
  handler,
});
};
```
produce:
```
...
schema:
  type: object
  required:
    - prop
  properties:
    prop:
      const: my-const
...
```
Swagger validator error message:
```
Structural error at paths./v1/users/.post.requestBody.content.application/json.schema.properties.prop
should NOT have additional properties
additionalProperty: const
```

### Expected Behavior

The swagger plugin will generate a valid swagger file, for example, enum with a single value when using json-schema `const` keyword

### PR DESCRIPTION
**Pull request by mikicho**
close #521 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/spec/openapi/utils.js [hunk 1]
```diff
@@ -84,6 +84,10 @@ function transformDefsToComponents (jsonSchema) {
     if (jsonSchema.patternProperties) {
       jsonSchema.additionalProperties = { type: 'string' }
       delete jsonSchema.patternProperties
+    } else if (jsonSchema.const) {
+      // Handle const, that is not part of OpenAPI definitions
```
Comment (Eomm) [inline lib/spec/openapi/utils.js:88]:
```suggestion
      // OAS 3.1 supports `const` but it is not supported by `swagger-ui`
      // https://swagger.io/docs/specification/data-models/keywords/
```

**Review by Eomm — CHANGES_REQUESTED** [review]
Could you add a unit test for this feature?

**Comment by mikicho** [comment]
@Eomm 
> Could you add a unit test for this feature?

Not sure we need to, `swagger/openapi should return valid swagger object` already testing it and it seems like the general approach is not test the transformer itself (the output..)

**Review by mcollina — COMMENTED** [review]
Can you please add a unit test?

File: lib/spec/swagger/utils.js [hunk 2]
```diff
@@ -164,6 +164,11 @@ function replaceUnsupported (jsonSchema) {
     if (jsonSchema.patternProperties) {
       jsonSchema.additionalProperties = { type: 'string' }
       delete jsonSchema.patternProperties
+    } else if (jsonSchema.const) {
+      // OAS 3.1 supports `const` but it is not supported by `swagger-ui`
+      // https://swagger.io/docs/specification/data-models/keywords/
```
Comment (Eomm) [inline lib/spec/swagger/utils.js:169]: The old comment, here in the `swagger` directory was fine.

- Swagger is an alias for OAS version 2 that does not support `const`
- The other comment, in `openapi` folder, is the file to support OAS version 3. So the comment there must be different from the following

```suggestion
      // Handle const, that is not part of OpenAPI definitions
```

**Review by Eomm — CHANGES_REQUESTED** [review]
We need a test that uses the `const` keyword within a schema and check that the output produced is the `enum` instead.

This use case is not covered by the existing tests

File: lib/spec/swagger/utils.js (hunk 2 above)
Comment (mikicho) [reply lib/spec/swagger/utils.js]: thanks, fixed

**Review by Eomm — APPROVED** [review]

**Comment by mikicho** [comment]
@Eomm can you please nerge it?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by darkgl0w — APPROVED** [review]
LGTM

**Pull request merged**